webui/react/src/hooks/useSettingsProvider.tsx

```diff
@@ -32,9 +33,10 @@ export const UserSettings = createContext<UserSettingsContext>({
 });
 
 export const SettingsProvider: React.FC<React.PropsWithChildren> = ({ children }) => {
-  const {
-    auth: { user, checked },
-  } = useStore();
+  const { user, checked } = Loadable.getOrElse(
+    { checked: true, isAuthenticated: false },
```

Comment:
This is exactly what I mean about the initState being confusing. We aren't using `isAuthenticated` in this file and the spinner should just check whether the Loadable is Loaded or not. The only value we actually want here is `user`.

This should be `const user = Loadable.map(useAuth().auth, auth => auth.user)` and then we can use that value in the component.

Reply:
This isn't a standard, it's just what certain parts of the code looks like. The information we want here is different than in other parts of the PR and so the code should look different.